core/utils.ts

```diff
@@ -219,3 +219,14 @@ export function checkExcessProperties<T>(
     );
   }
 }
+
+export function validateQueryString(session: Session, query: string) {
+  query = query.trim();
+  if (query.charAt(query.length - 1) === ";") {
```

Comment:
personally i like:

```suggestion
  if (query.trim().slice(-1)[0] === ";") {
```

but up to you